pkg/controllers/provisioningv2/rke2/bootstrap/controller.go

```diff
@@ -276,31 +272,58 @@ func (h *handler) OnChange(bootstrap *rkev1.RKEBootstrap, status rkev1.RKEBootst
 		result []runtime.Object
 	)
 
+	machine, err := rke2.GetOwnerCAPIMachine(bootstrap, h.machineCache)
+	if apierrors.IsNotFound(err) {
```

Comment:
This is fine the way you have it. I feel like it's more `Go style` for error handling but personally, I'd put the `IsNotFound` checks inside the `err != nil` blocks. That makes more logical sense but again I see the efficiency of short circuiting here